src/lib/statistics/statistics_objects/range_filter.cpp

```diff
@@ -136,56 +138,74 @@ std::unique_ptr<RangeFilter<T>> RangeFilter<T>::build_filter(const pmr_vector<T>
   }
 
   /*
-  * In case more than one value is present, first the elements are checked for potential overflows (e.g., when calculating
-  * the distince between INT::MIN() and INT::MAX(), the resulting distance might be to large for signed types).
+  * The code to determine the range boundaries requires calculating the distances between values. We cannot express the
+  * distance between -DBL_MAX and DBL_MAX using any of the standard data types. For these cases, the RangeFilter 
+  * effectively degrades to a MinMaxFilter (i.e., stores only a single range).
   * While being rather unlikely for doubles, it's more likely to happen when Opossum includes tinyint etc.
   * std::make_unsigned<T>::type would be possible to use for signed int types, but not for floating types.
-  * Approach: take the min and max values and simply check if the distance between both might overflow. In this case,
-  * fall back to a single range filter.
+  * Approach: take the min and max values and simply check if the distance between both might overflow.
   */
   const auto min_max = std::minmax_element(dictionary.cbegin(), dictionary.cend());
-  if ((*min_max.first < 0) &&
-      (*min_max.second > std::numeric_limits<T>::max() + *min_max.first)) {  // min_value is negative
+  if ((*min_max.first < 0) && (*min_max.second > std::numeric_limits<T>::max() + *min_max.first)) {
     return std::make_unique<RangeFilter<T>>(std::vector<std::pair<T, T>>{{*min_max.first, *min_max.second}});
   }
 
-  // calculate distances by taking the difference between two neighbouring elements
-  // vector stores <distance to next element, dictionary index>
+  // Running example:
+  // Input dictionary: 2 3 5 8 9 11
+  // max_ranges_count: 3
+  //
+  // Steps:
+  // 1. Calculate the distance from each value in the (sorted) dictionary to the next value and store a pair that
+  //    contains (a) the distance to the next value in the dictionary and (b) the entry's index in the dictionary:
+  //    (1,0) (2,1) (3,2) (1,3) (2,3)
+  // 2. Sort the vector in descending order by the size of the gaps (might be an unstable sort):
+  //    (3,2) (2,1) (2,3) (1,0) (1,3)
+  // 3. Shorten the vector to containt the `max_ranges_count - 1` biggest gaps
+  //    (3,2) (2,1)
+  // 4. Restore the original order of the dictionary by sorting on the second field
+  //    (2,1) (3,2) (0,5)
+  // 5. Add the highest value in the dictionary (more correctly, its index). We will not need the distance for this
```

Comment:
Too long ago. :D
I don't understand why we add `T{}` here and I don't see it in the example either.

Reply:
Ahh ... the added value is shown in the line before. And that makes sense.

Reply:
Good point. When I wrote the documentation, I had 4 and 5 in the wrong order. When I changed the text, I didn't update the example.